code/game/machinery/computer/guntracker.dm

```diff
@@ -0,0 +1,96 @@
+/obj/machinery/computer/guntracker
+	name = "firearm control console"
+	desc = "A console that can be used to track wireless-enabled firearms, and remotely control their activation."
+	icon_screen = "explosive"
+	light_color = LIGHT_COLOR_ORANGE
+	req_access = list(access_armory)
+	circuit = /obj/item/circuitboard/guntracker
+	var/id = 0.0
+	var/temp = null
+	var/status = 0
+	var/timeleft = 60
+	var/stop = 0.0
+	var/screen = 0 // 0 - No Access Denied, 1 - Access allowed
+
+/obj/machinery/computer/guntracker/attack_ai(var/mob/user)
+	return attack_hand(user)
+
+/obj/machinery/computer/guntracker/attack_hand(var/mob/user)
+	if(..())
+		return
+	ui_interact(user)
+
+/obj/machinery/computer/guntracker/ui_interact(var/mob/user)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if(!ui)
+		ui = new /datum/vueui/modularcomputer(user, src, "security-guntracker", 600, 400, "Firearm Control")
```

Comment:
Why you modular computer vueui type when on normal thing? this causes issues and things.

Reply:
It will be a modular computer shortly.